src/grids.jl

```diff
@@ -71,18 +71,29 @@ RegularCartesianGrid{Float32}
       domain (Lx, Ly, Lz) = (8.0f0, 8.0f0, 2.0f0)
 grid spacing (Δx, Δy, Δz) = (0.25f0, 0.25f0, 0.125f0)
 """
-function RegularCartesianGrid(T, N, L)
-    length(N) == 3 || throw(ArgumentError("N=$N must be a tuple of length 3."))
-    length(L) == 3 || throw(ArgumentError("L=$L must be a tuple of length 3."))
-
+function RegularCartesianGrid(T; N, x, y, z)
+    length(N) == 3        || throw(ArgumentError("N=$N must be a tuple of length 3."))
     all(isa.(N, Integer)) || throw(ArgumentError("N=$N should contain integers."))
-    all(isa.(L, Number))  || throw(ArgumentError("L=$L should contain numbers."))
-
-    all(N .>= 1) || throw(ArgumentError("N=$N must be nonzero and positive!"))
-    all(L .> 0)  || throw(ArgumentError("L=$L must be nonzero and positive!"))
-
+    all(N .>= 1)          || throw(ArgumentError("N=$N must be nonzero and positive!"))
+
+    function coord2xyz(c)
+        c == 1 && return "x"
+        c == 2 && return "y"
+        c == 3 && return "z"
+    end
+
+    for (i, c) in enumerate((x, y, z))
+        name = coord2xyz(i)
+        length(c) == 2       || throw(ArgumentError("$name=$c must be a tuple of length 2."))
```

Comment:
`x, y, z` can also be `Array`. I think you want to say something like `length($c) should be 2.`

Reply:
This wasn't changed.

Reply:
Sorry, just fixed this. Commit incoming.